### DISCUSSION TITLE
Failed to apply chaos: rpc error: code = Unknown desc = expected docker:// but got cri-o://3

### DISCUSSION THREAD

**Discussion by bsogithub** [chaos-mesh/chaos-mesh#3684] [Q&A]
## Bug Report

**What version of Kubernetes are you using?**
v1.21.11+31d53a1

**What version of Chaos Mesh are you using?**
2.3.2

**What did you do? / Minimal Reproducible Example**
Chaos-mesh was installed on Openshift 4.8.48 via helm. Two methods were used and both got the same error.
1) helm install chaos-mesh helm/chaos-mesh -n chaos-testing
2) helm install chaos-mesh helm/chaos-mesh -n chaos-testing  --set chaosDaemon.runtime=crio --set chaosDaemon.socketPath=/var/run/crio/crio.sock

This was the experiment to run:
kind: StressChaos
apiVersion: chaos-mesh.org/v1alpha1
metadata:
  namespace: robot-shop
  name: catalog-stress-test-4
spec:
  selector:
    namespaces:
      - robot-shop
    labelSelectors:
      service: catalogue
  mode: all
  stressors:
    memory:
      workers: 3
      size: 100%
    cpu:
      workers: 3
      load: 100
  duration: 2m

**What did you expect to see?**
The stresschaos ran on the catalogue pod successfully

**What did you see instead?**
Failed to apply chaos: rpc error: code = Unknown desc = expected docker:// but got cri-o://3

**Output of chaosctl**
Not quite sure what command to run but this is the output of stresschaos debug:
./chaosctl debug stresschaos -N chaos-testing -n robot-shop
failed to execute cmd:

        run command ps failed: failed get pid from pod robot-shop/catalogue-5dd69f5985-cww47: rpc error: code = Unknown desc = expected docker:// but got cri-o://3

**Comment by STRRL** [comment]
If seems that you are using CRI-O as the Kubernetes container runtime. 

Please follow the documentation, update the configuration for chaos mesh:

https://chaos-mesh.org/docs/production-installation-using-helm/#step-4-install-chaos-mesh-in-different-environments

**Comment by bsogithub** [comment]
@STRRL On my description, I did mentioned that I use the following command to install, I also did a reinstall again.
**helm install chaos-mesh helm/chaos-mesh -n chaos-testing**
WARNING: Kubernetes configuration file is group-readable. This is insecure. Location: /root/auth/kubeconfig
NAME: chaos-mesh
LAST DEPLOYED: Tue Oct  4 07:44:35 2022
NAMESPACE: chaos-testing
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
1. Make sure chaos-mesh components are running
   kubectl get pods --namespace chaos-testing -l app.kubernetes.io/instance=chaos-mesh

According to the doc (https://chaos-mesh.org/docs/production-installation-using-helm/#step-4-install-chaos-mesh-in-different-environments), using this install will default to using docker:
"
Docker
\# Default to \/var\/run\/docker.sock
helm install chaos-mesh chaos-mesh/chaos-mesh -n=chaos-testing --version 2.4.1
"

However, when I run the experiment, I am still getting:
Failed to apply chaos: rpc error: code = Unknown desc = expected docker:// but got cri-o://3

I think I did it according to documentation. Please let me know why I am still getting the error. Thank you!

**Comment by bsogithub** [comment]
@STRRL Please let me know how to install/config properly. Pointing me to the doc does not help because I was following it.

**Reply by STRRL** [reply]
You are using `contianerd` as the CRI, you should install with


```
helm install chaos-mesh chaos-mesh/chaos-mesh -n=chaos-mesh --set chaosDaemon.runtime=containerd --set chaosDaemon.socketPath=/run/containerd/containerd.sock --version 2.4.1
```
